src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1047,4 +1052,113 @@ private LegacySQLTypeName translateType(String datatype) {
             default: throw new IllegalArgumentException("Unknown datatype '" + datatype + "'");
         }
     }
+
+    private String externalTableName(String tableName, String suffix) {
+        return String.format("%s%s_%s", PDAO_EXTERNAL_TABLE_PREFIX, tableName, suffix);
+    }
+
+    public String createSoftDeleteExternalTable(Dataset dataset, String path, String tableName, String suffix) {
+        BigQueryProject bigQueryProject = bigQueryProjectForDataset(dataset);
+        String extTableName = externalTableName(tableName, suffix);
+        TableId tableId = TableId.of(prefixName(dataset.getName()), extTableName);
+        Schema schema = Schema.of(Field.of(PDAO_ROW_ID_COLUMN, LegacySQLTypeName.STRING));
+        ExternalTableDefinition tableDef = ExternalTableDefinition.of(path, schema, FormatOptions.csv());
+        TableInfo tableInfo = TableInfo.of(tableId, tableDef);
+        bigQueryProject.getBigQuery().create(tableInfo);
```

Comment:
Should we try to catch & inspect any exceptions raised here, so we can convert errors from a wildcard that matches 0 files into a 404 error? We do [this hack](https://github.com/DataBiosphere/jade-data-repo/blob/develop/src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java#L455-L457) when using a load job, not sure what the equivalent would be for an external table definition.

Reply:
Thanks for the pointer, yes I'll add this here and also move the missing non-wildcard file check here like you mention in the previous comment.